### ISSUE TITLE
Generator Doesn't Work

### ISSUE THREAD

**Issue by zackerydev** [oclif/oclif#829]
Hi All!

Been developing a new library for a few days now and I'm having trouble getting the `npm pack` and publishing working.

This happens for me on a freshly generated cli:
```
~/GitHub/oclif oclif generate mynewcli2

     _-----_
    |       |    ╭──────────────────────────╮
    |--(o)--|    │  Time to build an oclif  │
   `---------´   │    CLI! Version: 2.4.4   │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |
   __'.___.'__
 ´   `  |° ´ Y `

Cloning into '/Users/zack.griesinger/GitHub/oclif/mynewcli2'...
? npm package name mynewcli2
? command bin name the CLI will export mnc2
? description oclif example Hello World CLI
? author Zackery Griesinger @zgriesinger
? license MIT
? Who is the GitHub owner of repository (https://github.com/OWNER/repo) zgriesinger
? What is the GitHub name of repository (https://github.com/owner/REPO) hello-world
? Select a package manager yarn
    force mynewcli2/package.json
    force mynewcli2/.gitignore

Changes to package.json were detected.

Running yarn install for you to install the required dependencies.
yarn install v1.22.17
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
warning oclif > aws-sdk > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning oclif > aws-sdk > uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
warning oclif > aws-sdk > url > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
warning "oclif > yeoman-environment@3.9.1" has unmet peer dependency "mem-fs@^1.2.0 || ^2.0.0".
warning "oclif > yeoman-environment@3.9.1" has unmet peer dependency "mem-fs-editor@^8.1.2 || ^9.0.0".
[5/5] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 7.36s.
yarn run v1.22.17
$ shx rm -rf dist && tsc -b
✨  Done in 2.00s.
(node:37711) [MODULE_NOT_FOUND] ModuleLoadError Plugin: mynewcli2: [MODULE_NOT_FOUND] require failed to load /Users/zack.griesinger/GitHub/oclif/mynewcli2/src/commands/hello
module: @oclif/core@1.3.4
task: toCached
plugin: mynewcli2
root: /Users/zack.griesinger/GitHub/oclif/mynewcli2
See more details with DEBUG=*
(Use `node --trace-warnings ...` to show where the warning was created)
(node:37711) [MODULE_NOT_FOUND] ModuleLoadError Plugin: mynewcli2: [MODULE_NOT_FOUND] require failed to load /Users/zack.griesinger/GitHub/oclif/mynewcli2/src/commands/hello/world
module: @oclif/core@1.3.4
task: toCached
plugin: mynewcli2
root: /Users/zack.griesinger/GitHub/oclif/mynewcli2
See more details with DEBUG=*
replacing <!-- usage --> in README.md
replacing <!-- commands --> in README.md
replacing <!-- toc --> in README.md

Created mynewcli2 in /Users/zack.griesinger/GitHub/oclif/mynewcli2
```

Without doing anything i'm getting `MODULE_NOT_FOUND` issues. After running this I can't do any of the `./bin/run` commands, but `./bin/dev` still works.  I've tried numerous times with everything I can think of, if anyone has any ideas that would be great.

**Comment by mdonnalley** [comment]
@zgriesinger I'm not able to reproduce this issue on 2.4.4

Are you able to provide any more detail about how to recreate it?

**Comment by zgriesinger** [comment]
I'm not sure what I'm doing wrong.

On Node `v16.13.2` on an M1 Macbook (if that matters? FWIW i tested on a different machine and got the same issue)

Running:
```bash
npm i -g oclif
oclif --help # prints 2.4.4
oclif generate mynewcli
```
Those are the commands I'm running and I get the MODULE_NOT_FOUND error. Not sure what else I can try or how to recreate.

**Comment by jaredcosulich** [comment]
I'm seeing what appears to be a different error when I attempt to run the generator.

```
$ npx oclif generate nodcli

     _-----_     
    |       |    ╭──────────────────────────╮
    |--(o)--|    │  Time to build an oclif  │
   `---------´   │    CLI! Version: 2.4.5   │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

Cloning into '/Users/jaredcosulich/workspace/nod/nodcli'...
? npm package name nodcli
? command bin name the CLI will export nod
? description oclif example Hello World CLI
? author Jared Cosulich @jaredcosulich
? license MIT
? Who is the GitHub owner of repository (https://github.com/OWNER/repo) jaredcosulich
? What is the GitHub name of repository (https://github.com/owner/REPO) nodcli
? Select a package manager yarn
    force nodcli/package.json
    force nodcli/.gitignore

Changes to package.json were detected.

Running yarn install for you to install the required dependencies.
yarn install v1.22.17
warning ../package.json: No license field
[1/4] 🔍  Resolving packages...
warning oclif > aws-sdk > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning oclif > aws-sdk > url > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning oclif > aws-sdk > uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "oclif > yeoman-environment@3.9.1" has unmet peer dependency "mem-fs@^1.2.0 || ^2.0.0".
warning "oclif > yeoman-environment@3.9.1" has unmet peer dependency "mem-fs-editor@^8.1.2 || ^9.0.0".
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
✨  Done in 21.98s.
yarn run v1.22.17
warning ../package.json: No license field
$ shx rm -rf dist && tsc -b
../../node_modules/@types/babel__generator/index.d.ts:11:20 - error TS2307: Cannot find module '@babel/types' or its corresponding type declarations.

11 import * as t from '@babel/types';
                      ~~~~~~~~~~~~~~

../../node_modules/@types/babel__template/index.d.ts:9:31 - error TS2307: Cannot find module '@babel/parser' or its corresponding type declarations.

9 import { ParserOptions } from '@babel/parser';
                                ~~~~~~~~~~~~~~~

../../node_modules/@types/babel__template/index.d.ts:10:48 - error TS2307: Cannot find module '@babel/types' or its corresponding type declarations.

10 import { Expression, Program, Statement } from '@babel/types';
                                                  ~~~~~~~~~~~~~~

../../node_modules/@types/babel__traverse/index.d.ts:13:20 - error TS2307: Cannot find module '@babel/types' or its corresponding type declarations.

13 import * as t from '@babel/types';
                      ~~~~~~~~~~~~~~

../../node_modules/@types/babel__traverse/index.d.ts:63:18 - error TS2411: Property 'blacklist' of type 'any[] | undefined' is not assignable to 'string' index type '(VisitNode<S, any> & VisitNode<S, t.Aliases>) | undefined'.

63 export interface TraverseOptions<S = Node> extends Visitor<S> {
                    ~~~~~~~~~~~~~~~

../../node_modules/@types/babel__traverse/index.d.ts:63:18 - error TS2411: Property 'denylist' of type 'any[] | undefined' is not assignable to 'string' index type '(VisitNode<S, any> & VisitNode<S, t.Aliases>) | undefined'.

63 export interface TraverseOptions<S = Node> extends Visitor<S> {
                    ~~~~~~~~~~~~~~~

../../node_modules/@types/babel__traverse/index.d.ts:64:5 - error TS2411: Property 'scope' of type 'Scope | undefined' is not assignable to 'string' index type '(VisitNode<S, any> & VisitNode<S, t.Aliases>) | undefined'.

64     scope?: Scope | undefined;
       ~~~~~

../../node_modules/@types/babel__traverse/index.d.ts:65:5 - error TS2411: Property 'noScope' of type 'boolean | undefined' is not assignable to 'string' index type '(VisitNode<S, any> & VisitNode<S, t.Aliases>) | undefined'.

65     noScope?: boolean | undefined;
       ~~~~~~~

../../node_modules/@types/babel__traverse/index.d.ts:248:69 - error TS2536: Type '"type"' cannot be used to index type 'T'.

248     type: T extends null | undefined ? undefined : T extends Node ? T['type'] : string | undefined;
                                                                        ~~~~~~~~~

../../node_modules/@types/babel__core/index.d.ts:12:31 - error TS2307: Cannot find module '@babel/parser' or its corresponding type declarations.

12 import { ParserOptions } from '@babel/parser';
                                 ~~~~~~~~~~~~~~~

../../node_modules/@types/babel__core/index.d.ts:15:20 - error TS2307: Cannot find module '@babel/types' or its corresponding type declarations.

15 import * as t from '@babel/types';
                      ~~~~~~~~~~~~~~

../../node_modules/@types/jest/index.d.ts:35:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'beforeEach' must be of type 'HookFunction', but here has type 'Lifecycle'.

35 declare var beforeEach: jest.Lifecycle;
               ~~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2624:13
    2624 declare var beforeEach: Mocha.HookFunction;
                     ~~~~~~~~~~
    'beforeEach' was also declared here.

../../node_modules/@types/jest/index.d.ts:37:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'afterEach' must be of type 'HookFunction', but here has type 'Lifecycle'.

37 declare var afterEach: jest.Lifecycle;
               ~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2642:13
    2642 declare var afterEach: Mocha.HookFunction;
                     ~~~~~~~~~
    'afterEach' was also declared here.

../../node_modules/@types/jest/index.d.ts:38:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'describe' must be of type 'SuiteFunction', but here has type 'Describe'.

38 declare var describe: jest.Describe;
               ~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2658:13
    2658 declare var describe: Mocha.SuiteFunction;
                     ~~~~~~~~
    'describe' was also declared here.

../../node_modules/@types/jest/index.d.ts:40:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'xdescribe' must be of type 'PendingSuiteFunction', but here has type 'Describe'.

40 declare var xdescribe: jest.Describe;
               ~~~~~~~~~

  node_modules/@types/mocha/index.d.ts:2679:13
    2679 declare var xdescribe: Mocha.PendingSuiteFunction;
                     ~~~~~~~~~
    'xdescribe' was also declared here.

../../node_modules/@types/jest/index.d.ts:41:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'it' must be of type 'TestFunction', but here has type 'It'.

41 declare var it: jest.It;
               ~~

  node_modules/@types/mocha/index.d.ts:2693:13
    2693 declare var it: Mocha.TestFunction;
                     ~~
    'it' was also declared here.

../../node_modules/@types/jest/index.d.ts:43:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'xit' must be of type 'PendingTestFunction', but here has type 'It'.

43 declare var xit: jest.It;
               ~~~

  node_modules/@types/mocha/index.d.ts:2714:13
    2714 declare var xit: Mocha.PendingTestFunction;
                     ~~~
    'xit' was also declared here.

../../node_modules/@types/jest/index.d.ts:44:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'test' must be of type 'TestFunction', but here has type 'It'.

44 declare var test: jest.It;
               ~~~~

  node_modules/@types/mocha/index.d.ts:2707:13
    2707 declare var test: Mocha.TestFunction;
                     ~~~~
    'test' was also declared here.

../../node_modules/@types/jest/index.d.ts:490:51 - error TS2307: Cannot find module 'jest-diff' or its corresponding type declarations.

490             diff(a: any, b: any, options?: import("jest-diff").DiffOptions): string | null;
                                                      ~~~~~~~~~~~

../../node_modules/@types/jest/index.d.ts:544:44 - error TS2307: Cannot find module 'pretty-format' or its corresponding type declarations.

544     type SnapshotSerializerPlugin = import('pretty-format').Plugin;
                                               ~~~~~~~~~~~~~~~


Found 20 errors.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
/Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/yeoman-generator/node_modules/execa/lib/error.js:59
		error = new Error(message);
		        ^

Error: Command failed with exit code 1: yarn run build
    at makeError (/Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/yeoman-generator/node_modules/execa/lib/error.js:59:11)
    at Function.module.exports.sync (/Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/yeoman-generator/node_modules/execa/index.js:188:17)
    at CLI.spawnCommand.spawnCommandSync (/Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/yeoman-generator/lib/actions/spawn-command.js:35:16)
    at CLI.end (/Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/oclif/lib/generators/cli.js:144:14)
    at Object.<anonymous> (/Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/yeoman-generator/lib/index.js:1097:23)
    at /Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/run-async/index.js:49:25
    at new Promise (<anonymous>)
    at /Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/run-async/index.js:26:19
    at /Users/jaredcosulich/.npm/_npx/3f625c11f20f0be4/node_modules/yeoman-generator/lib/index.js:1098:9
    at new Promise (<anonymous>) {
  shortMessage: 'Command failed with exit code 1: yarn run build',
  command: 'yarn run build',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

```

**Comment by jlvandenhout** [comment]
Similar issue here. Running `yarn dlx oclif generate cli` gives me:
```
➤ YN0000: ┌ Resolution step
➤ YN0061: │ querystring@npm:0.2.0 is deprecated: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
➤ YN0061: │ uuid@npm:3.3.2 is deprecated: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
➤ YN0000: └ Completed in 5s 227ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ xml2js@npm:0.4.19 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ xmlbuilder@npm:9.0.7 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yeoman-environment@npm:3.9.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yeoman-generator@npm:5.6.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yosay@npm:2.0.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 10s 452ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed in 0s 889ms
➤ YN0000: Done with warnings in 16s 681ms

 ›   ModuleLoadError: [MODULE_NOT_FOUND] require failed to load 
 ›   /home/develop/.yarn/berry/cache/oclif-npm-2.6.0-7be1d4e51b-8.zip/node_modules/oclif/lib/commands/generate
 ›   Code: MODULE_NOT_FOUND
```

**Comment by jackharrhy** [comment]
Dirty fix, but if you want to simply silence type errors coming from `node_modules`, you can add:

```json
{
  "compilerOptions": {
    "skipLibCheck": true
  },
}
```

to your tsconfig.json.

After adding the above, `npm run build` completes without error for me.

**Comment by rtviii** [comment]
I encountered this issue as well AFTER having successfully built the cli once with `npx oclif pack deb`. The next day i made some edits to it and tried to rebuild only to get:
```
yarn run v1.22.17
$ /home/rxz/dev/docker_ribxz/cli/node_modules/.bin/oclif pack deb
 ›   ModuleLoadError: [MODULE_NOT_FOUND] require failed to load /home/rxz/dev/docker_ribxz/cli/node_modules/oclif/lib/commands/pack/deb.js: Cannot find module
 ›    'node:crypto'
 ›   Require stack:
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/oclif/lib/util.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/oclif/lib/log.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/oclif/lib/tarballs/build.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/oclif/lib/tarballs/index.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/oclif/lib/commands/pack/deb.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/@oclif/core/lib/module-loader.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/@oclif/core/lib/config/plugin.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/@oclif/core/lib/config/config.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/@oclif/core/lib/config/index.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/@oclif/core/lib/command.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/@oclif/core/lib/index.js
 ›   - /home/rxz/dev/docker_ribxz/cli/node_modules/oclif/bin/run
 ›   Code: MODULE_NOT_FOUND
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by rtviii** [comment]
Make sure you `node` version is latest. This fixed the issue for me (failed via above on v14.12.x, but worked on v18.x.x).

**Comment by mdonnalley** [comment]
This issue has gone quiet and it looks like most people were seeing this on older versions of oclif. I'm going to close this issue but feel free to open a new issue if it's still happening. Thanks!
